api_types.ts

```diff
@@ -339,7 +343,11 @@ type TypeOfMap<T extends UnionType> = T extends Type
  * the parameter defintion for that formula.
  */
 export type ParamValues<ParamDefsT extends ParamDefs> = {
-  [K in keyof ParamDefsT]: ParamDefsT[K] extends ParamDef<infer T> ? TypeOfMap<T> : never;
+  [K in keyof ParamDefsT]: ParamDefsT[K] extends OptionalParamDef<infer S>
+    ? TypeOfMap<S> | undefined
```

Comment:
This is the heart of the change. The rest of it is getting TypeScript to get enough type inference from the return value of `makeParameter()` such that this check works--that it understands a parameter definition could possibly extend OptionalParamDef.